src/main/java/core/basesyntax/strategy/SupplyOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> stock) {
+        stock.merge(transaction.getFruit(), transaction.getQuantity(), Integer::sum);
```

This violates checklist guidance: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance." The handler does not validate inputs. Add defensive checks at the start of `handle(...)` and throw informative RuntimeException messages when preconditions fail:
- `transaction` must not be null
- `stock` must not be null
Validate these and fail fast to avoid NPEs later.